libvast/vast/system/reader_command.hpp

```diff
@@ -70,21 +71,35 @@ caf::message reader_command(const command& cmd, caf::actor_system& sys,
     else
       file = std::string{Reader::defaults::path};
   }
-  // Supply an alternate schema, if requested.
-  expected<vast::schema> schema{caf::none};
+  // Get the default schema from the registry.
+  auto schema = event_types::get();
+  // Update with an alternate schema, if requested.
+  vast::schema reader_schema;
   {
     auto sc = caf::get_if<std::string>(&options, category + ".schema");
     auto sf = caf::get_if<std::string>(&options, category + ".schema-file");
     if (sc && sf)
       return make_message(
         make_error(ec::invalid_configuration,
                    "had both schema and schema-file provided"));
-    if (sc)
-      schema = to<vast::schema>(*sc);
-    if (sf)
-      schema = load_schema_file(*sf);
-    if (!schema && schema.error() != caf::none)
-      return make_message(std::move(schema.error()));
+    auto update = [&]() -> caf::expected<vast::schema> {
+      if (sc)
+        return to<vast::schema>(*sc);
+      if (sf)
+        return load_schema_file(*sf);
+      return caf::no_error;
+    }();
+    if (update) {
+      if (!schema || schema->empty()) {
+        reader_schema = *update;
+      } else {
+        reader_schema = *schema;
+        reader_schema.update(*update);
+      }
+      schema = &reader_schema;
+    }
+    if (!update && update.error() != caf::no_error)
```

Comment:
```suggestion
   else  if (update.error() != caf::no_error)
```

Plus additional braces.